packages/app-store/_utils/useAddAppMutation.ts

```diff
@@ -57,9 +57,12 @@ function useAddAppMutation(_type: App["type"] | null, allOptions?: UseAddAppMuta
             location.search
           ),
       ...(type === "google_calendar" && { installGoogleVideo: options?.installGoogleVideo }),
+      ...(variables && variables.teamId && { teamId: variables.teamId }),
     };
     const stateStr = encodeURIComponent(JSON.stringify(state));
-    const searchParams = `?state=${stateStr}`;
+    const searchParams = `?state=${stateStr}${
+      variables && variables.teamId ? `&teamId=${variables.teamId}` : ""
+    }`;
```

Comment:
More readable and cleaner imho 

```suggestion
const searchParams = new URLSearchParams({
  state: stateStr,
  ...(variables && variables.teamId && { teamId: variables.teamId }),
}).toString();
```